src/CMakeLists.txt

```diff
@@ -325,12 +329,13 @@ target_link_libraries(${src_executable_hotgym}
     ${core_library}
     ${COMMON_OS_LIBS}
 )
+
     
 # for future reference; to link to shared library, (Not on MSVC)
 #          link with ${src_lib_shared} rather than ${core_library}
                   
 target_compile_options( ${src_executable_hotgym} PUBLIC ${INTERNAL_CXX_FLAGS})
-target_compile_definitions(${src_executable_hotgym} PRIVATE ${COMMON_COMPILER_DEFINITIONS})
+target_compile_definitions(${src_executable_hotgym} PRIVATE ${COMMON_COMPILER_DEFINITIONS} -DNTA_LIBC_MUSL=${NTA_LIBC_MUSL})
```

Comment:
why isn't enough specifying this in COMMON_COMPILER_DEFS above, but I have to pass it manually here?

Reply:
Hmmm, if you put it in COMMON_COMPILER_DEFS it should not need to be manually added.  Something must not be right someplace, but I did a quick look and could not find a problem.

Reply:
COMMON_COMPILER_DEFS  seems to be empty in src/CMakeLists, but it is set in CommonCompiler.cmake. One possibility is that the _STR version is passed further along? 

Anyway, when using `option()` as you've suggested above it works as expected and I can remove the extra ifs.